app/user.go

```diff
@@ -2264,15 +2266,61 @@ func (a *App) ConvertBotToUser(bot *model.Bot, userPatch *model.UserPatch, sysad
 }
 
 func (a *App) GetThreadsForUser(userID, teamID string, options model.GetUserThreadsOpts) (*model.Threads, *model.AppError) {
-	threads, err := a.Srv().Store.Thread().GetThreadsForUser(userID, teamID, options)
-	if err != nil {
+	var result model.Threads
+	var eg errgroup.Group
+
+	eg.Go(func() error {
```

Comment:
I am wondering if we should wait (or perhaps coordinate) with the `ThreadsOnly` PR https://github.com/mattermost/mattermost-server/pull/19349? This will slightly change the semantics of that PR. 0/5 from my side, I just want both PRs to get in with minimal re-work.

Reply:
Agreed, @isacikgoz, should we coordinate more closely on this front? Adding `ThreadsOnly` should be trivial on top of these changes, but I also know we discussed possibly reworking those changes to expose a new endpoint instead (which would still be easy with these changes).

Reply:
Did you mean to mention someone else? :wink: I don't think Ibrahim is working on this.

Reply:
Haha, yes, I'm mixing up my work on CRT with my work with @isacikgoz yesterday on the mobile outage :D -- apologies, @isacikgoz! cc @ashishbhate :)

Reply:
@lieut-data this PR lays the ground work for splitting up things up even further. Lets get this PR in and I can rework the other https://github.com/mattermost/mattermost-server/pull/19349 and go from there.